core/eolearn/tests/test_core_tasks.py

```diff
@@ -214,49 +215,54 @@ def test_duplicate_feature(patch):
-    assert patch.mask["test"].shape == shape
-    assert np.array_equal(patch.mask["test"], compare_data)
-
-    with pytest.raises(ValueError):
-        # Expected a ValueError when trying to initialize a feature with a wrong shape dimensions.
-        patch = InitializeFeatureTask((FeatureType.MASK_TIMELESS, "wrong"), shape=shape, init_value=init_val)(patch)
-
-    init_val = 123
-    shape = (10, 10, 3)
+@pytest.mark.parametrize(
+    "init_val, shape, feature, feature_wrong",
+    [
+        (123, (5, 10, 10, 3), (FeatureType.MASK, "test"), (FeatureType.MASK_TIMELESS, "wrong")),
+        (123, (10, 10, 3), (FeatureType.MASK_TIMELESS, "test"), (FeatureType.MASK, "wrong")),
+    ],
+)
+def test_initialize_feature(
+    init_val: float, shape: Tuple[int, ...], feature: FeatureSpec, feature_wrong: FeatureSpec, patch: EOPatch
+) -> None:
     compare_data = np.ones(shape) * init_val
 
-    patch = InitializeFeatureTask((FeatureType.MASK_TIMELESS, "test"), shape=shape, init_value=init_val)(patch)
-    assert patch.mask_timeless["test"].shape == shape
-    assert np.array_equal(patch.mask_timeless["test"], compare_data)
+    patch = InitializeFeatureTask(feature, shape=shape, init_value=init_val)(patch)
+    assert patch[feature].shape == shape
+    assert_array_equal(patch[feature], compare_data)
 
     with pytest.raises(ValueError):
         # Expected a ValueError when trying to initialize a feature with a wrong shape dimensions.
-        patch = InitializeFeatureTask((FeatureType.MASK, "wrong"), shape=shape, init_value=init_val)(patch)
-
-    init_val = 123
-    shape = (5, 10, 10, 3)
-    compare_data = np.ones(shape) * init_val
-    new_names = ("F1", "F2", "F3")
-
-    patch = InitializeFeatureTask({FeatureType.MASK: new_names}, shape=shape, init_value=init_val)(patch)
-    assert set(new_names) < set(patch.mask), "Failed to initialize new features from a shape tuple."
-    assert all(patch.mask[key].shape == shape for key in new_names)
-    assert all(np.array_equal(patch.mask[key], compare_data) for key in new_names)
+        InitializeFeatureTask(feature_wrong, shape=shape, init_value=init_val)(patch)
 
-    patch = InitializeFeatureTask({FeatureType.DATA: new_names}, shape=(FeatureType.DATA, "bands"))(patch)
-    assert set(new_names) < set(patch.data), "Failed to initialize new features from an existing feature."
-    assert all(patch.data[key].shape == patch.data["bands"].shape for key in new_names)
 
+@pytest.mark.parametrize(
+    "init_val, shape, feature_type, new_names, case_feature_spec",
+    [
+        (123, (5, 10, 10, 3), FeatureType.MASK, ("F1", "F2", "F3"), False),
+        (123, (FeatureType.DATA, "bands"), FeatureType.DATA, ("F1", "F2", "F3"), True),
+    ],
+)
+def test_initialize_feature_rename(
+    init_val: float,
+    shape: Union[Tuple[int, ...], FeatureSpec],
+    feature_type: FeatureType,
+    new_names: Tuple[str, ...],
+    patch: EOPatch,
+    case_feature_spec: bool,
```

Comment:
i needed flag for if result_shape is need to be extracted i did not want to use
```py
isinstance(shape, tuple) and len(shape) in (3, 4) and all(isinstance(x, int) for x in shape):
```
looking for better solution

Reply:
separate into two tests?

Reply:
done

Reply:
you can now simplify the types. After that you might be able to have less lines (put all parameters in one line and rerun black to see if it could work)

Reply:
done